articles/active-directory/develop/quickstart-v2-java-webapp.md

```diff
@@ -33,86 +32,124 @@ When you've completed the guide, your application will accept sign-ins of person
+    1. Find `aad.clientId` and update the value of `Enter_the_Application_Id_here` with the **Application (client) ID** value
+    of the application you registered.
+    1. Find `aad.authority` and update the value of `Enter_the_Tenant_Name_Here` with the **Directory (tenant) ID** value of
+    the application you registered.
+    1. Find `aad.secretKey` and update the value of `Enter_the_Client_Secret_Here` with the **Client Secret** you created in
+    **Certificates & Secrets** for the application you registered.
 
 > [!div renderon="docs"]
 > Where:
 >
 > - `Enter_the_Application_Id_here` - is the Application Id for the application you registered.
-> - `Enter_the_Client_Secret_Here` - is the **Client Secret** you created in **Certificates & Secrets** for the application you registered.
+> - `Enter_the_Client_Secret_Here` - is the **Client Secret** you created in **Certificates & Secrets** for the application
+you registered.
 
 #### Step 4: Run the code sample
-1. Run the code sample, and open a browser and navigate to *http://localhost:8080*.
-1. The front page contains a **sign-in** button. Click on the **sign-in** button to redirect to Azure Active Directory. The user will be prompted for their credentials.  
-1. After successfully authenticating on Azure Active Directory, they will be redirected to *http://localhost:8080/msal4jsamples/secure/aad*. They are officially signed in to the application, and the page should show information for the signed in account. It will also contain buttons for: 
+
+To run the project, you can either:
+
+- Run it directly from your IDE by using the embedded spring boot server
+- or package it to a WAR file using [maven](https://maven.apache.org/plugins/maven-war-plugin/usage.html) and deploy it to a J2EE container solution such as [Apache Tomcat](https://tomcat.apache.org/maven-plugin-trunk/tomcat6-maven-plugin/examples/deployment.html).
+
+##### Running from IDE
+
+1.If you are running the web application from an IDE, click on run, then navigate to the home page of the project. For this sample, the standard home page URL is <http://localhost:8080>
+
+1. The front page contains a **sign-in** button. Click on the **sign-in** button to redirect to Azure Active Directory. The
+user will be prompted for their credentials.  
+1. After successfully authenticating on Azure Active Directory, they will be redirected to
+*http://localhost:8080/msal4jsamples/secure/aad*. They are officially signed in to the application, and the page should show
+information for the signed in account. It will also contain buttons for:
     - *Sign Out*: Will sign out the current user from the application, and redirect them the home page.
-    - *Show Users*: Will acquire a token for the Microsoft Graph, then call the Microsoft Graph with the token attached to the request to get all of the users in the tenant.
+    - *Show User Info*: Will acquire a token for the Microsoft Graph, then call the Microsoft Graph with the token attached to
```

Comment:
Perhaps: "Acquires a token for Microsoft Graph and calls Microsoft Graph with a request containing the token. Returns information about the signed-in user."

Reply:
looks good, changed the statement.